crates/sui-graphql-rpc/src/data.rs

```diff
@@ -21,6 +24,10 @@ pub(crate) type Conn<'c> = <Db as QueryExecutor>::DbConnection<'c>;
 pub(crate) type DieselConn = <Db as QueryExecutor>::Connection;
 pub(crate) type DieselBackend = <Db as QueryExecutor>::Backend;
 
+/// Helper types for accessing a shared `DataLoader` instance.
+#[derive(Clone)]
+pub(crate) struct DataLoader(pub Arc<AGDataLoader<Db>>);
```

Comment:
Had to create this alias so that I could wrap the data loader in an `Arc` (so it could be shared between the package resolver and the GraphQL context).

Reply:
is this more preferable over having to refer to it as `Arc<DataLoader>` in the package resolver?

Reply:
The benefit of this newtype is that for all the places where we fetch the DataLoader from the request context, we don't need to remember what it's wrapped in. For `PackageResolver` itself, it's fine to just have an `Arc<DataLoader>` sitting there, but seeing as the alias exists, I thought I may as well use it there too.